### DISCUSSION TITLE
How to use dynamic model

### DISCUSSION THREAD

**Discussion by aramis-it** [prisma/prisma#4384] [Q&A] [answered]
```ts
app.post('/api/:controller', async (req, res) => {

  const result = await prisma[model(req.params["controller"])].create({
    data: {
      ...req.body
    },
  })
  res.json(req.params[0])
})

const model = (params: any): Prisma.ModelName => {
  return params ;
}
```

Error

```
src/index.ts:20:63 - error TS2349: This expression is not callable.
  Each member of the union type '(<T extends userCreateArgs>(args: Subset<T, userCreateArgs>) => CheckSelect<T, Prisma__userClient<user>, Prisma__userClient<userGetPayload<T, keyof T>>>) | ... 
13 more ... | (<T extends cardurlCreateArgs>(args: Subset<...>) => CheckSelect<...>)' has signatures, but none of those signatures are compatible with each other.

20   const result = await prisma[type(req.params["controller"])].create({
```

**Comment by ryands17** [comment, marked as answer]
Hey @ulziibadrakh-p 👋 

Dynamic model is not recommended to use as the parameters for each models are different along with their TS types. It would be better to write separate functions for models to get better validation and type support.

If you still want to use Dynamic models, you would need to use the `any` type as TypeScript wouldn't be able to understand the type that you are trying to use for a given model.

**Reply by irving-caamal** [reply]
it means:

use it at your own risk

**Reply by platform-kit** [reply]
@irvv17 @ryands17 

I have a use case where I will need to use a model determined dynamically at runtime. I understand it's not best practice according to your ethos, but could you point me in the direction of how I might make this happen?

I'm currently a bit baffled, as I'm coming from es6 where I could reference a property dynamically - obviously this is not the case in TS. 

So please advise - how would I go about specifying a model via a variable?

**Reply by ryands17** [reply]
Currently it's not straightforward, but you can read this request and some workarounds.

https://github.com/prisma/prisma/issues/5273

We will be implementing a better wrapper in the future.

**Reply by platform-kit** [reply]
@ryands17 Thanks. I added a bit of code I cooked up to address this issue in #7075

**Comment by aramis-it** [comment]
Okay, thanks for your reply.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
